workitem/workitem_repository.go

```diff
@@ -678,48 +681,151 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
+				Total:       0,
+				Closed:      0,
+			}
+		}
+	}
+
+	type IterationHavingChildrenID struct {
+		Children    string `gorm:"column:children"`
+		IterationID string `gorm:"column:iterationid"`
+	}
+	var itrChildren []IterationHavingChildrenID
+	queryIterationWithChildren := fmt.Sprintf(`
+	WITH PathResolver AS
+	(SELECT CASE
+				WHEN path = '' THEN replace(id::text, '-', '_')::ltree
+				ELSE concat(path::text, '.', REPLACE(id::text, '-', '_'))::ltree
+			END AS pathself,
+			id
+	FROM %s)
+	SELECT array_agg(iterations.id)::text AS children,
+		PathResolver.id::text AS iterationid
+	FROM %s,
+		PathResolver
+	WHERE path <@ PathResolver.pathself
+	AND space_id = '%s'
+	GROUP BY (PathResolver.pathself,
+          PathResolver.id)`,
+		iterationTableName,
+		iterationTableName,
+		spaceID.String())
+	db = r.db.Raw(queryIterationWithChildren)
+	db.Scan(&itrChildren)
+	childMap := map[string][]string{}
+	for _, r := range itrChildren {
+		// Following can be done by implementing Valuer interface for type IterationHavingChildrenID
+		r.Children = strings.TrimPrefix(r.Children, "{")
+		r.Children = strings.TrimRight(r.Children, "}")
+		children := strings.Split(r.Children, ",")
+		childMap[r.IterationID] = children
+	}
 	countsMap := map[string]WICountsPerIteration{}
-	for _, iterationWithCount := range res {
-		countsMap[iterationWithCount.IterationID] = iterationWithCount
+	for _, i := range wiMap {
+		t := i.Total
+		c := i.Closed
+		if children, exists := childMap[i.IterationID]; exists {
+			for _, child := range children {
+				if _, exists := wiMap[child]; exists {
+					t += wiMap[child].Total
+					c += wiMap[child].Closed
+				}
+			}
+		}
+		countsMap[i.IterationID] = WICountsPerIteration{
+			IterationID: i.IterationID,
+			Total:       t,
+			Closed:      c,
+		}
 	}
 	return countsMap, nil
 }
 
 // GetCountsForIteration returns Closed and Total counts of WI for given iteration
 // It executes
 // SELECT count(*) as Total, count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed FROM "work_items" where fields@> concat('{"system.iteration": "%s"}')::jsonb and work_items.deleted_at is null
-func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, iterationID uuid.UUID) (map[string]WICountsPerIteration, error) {
+func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, itr *iteration.Iteration) (map[string]WICountsPerIteration, error) {
 	var res WICountsPerIteration
+	pathOfIteration := append(itr.Path, itr.ID)
+	// get child IDs of the iteration
+	var childIDs []uuid.UUID
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	getIterationsOfSpace := fmt.Sprintf(`select id from %s where path <@ '%s'`, iterationTableName, pathOfIteration.Convert())
+	db := r.db.Raw(getIterationsOfSpace)
+	db = db.Debug()
+	db.Pluck("id", &childIDs)
+	if db.Error != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
```

Comment:
yup added error logs in https://github.com/almighty/almighty-core/pull/1151/commits/19150ea07ffc361bfbcfc2cb60e7c2ca40c26c9c